lib/chef/knife/azure_server_create.rb

```diff
@@ -957,9 +955,7 @@ def create_server_def
             server_def[:azure_domain_name] = $2
             server_def[:azure_domain_user] = $1
           else
-            # Format error.
-            ui.error("Format error for --azure-domain-user option. Supported format are user principal name (UPN) format (user@fully-qualified-DNS-domain) or the fully-qualified-DNS-domain\\username format")
-            exit 1
+            server_def[:azure_domain_user] = locate_config_value(:azure_domain_user)
```

Comment:
Is our way of constructing the username that we pass to Azure going to allow us to specify a user that's not in the domain we're trying to join? Users from one domain can be authorized to join systems to another domain if there is a trust between those domains.